src/datasets/load.py

```diff
@@ -305,22 +457,52 @@ def prepare_module(
     else:
         main_folder_path = force_local_path
 
-    # We have three ways to find the processing file:
-    # - if os.path.join(path, name) is a file or a remote url
-    # - if path is a file or a remote url
-    # - otherwise we assume path/name is a path to our S3 bucket
-    combined_path = path if path.endswith(name) else os.path.join(path, name)
-
-    if os.path.isfile(combined_path):
+    # We have several ways to find the processing file:
+    # - if os.path.join(path, name) is a local python file
+    #   -> use the module from the python file
+    # - if path is a local directory (but no python file)
+    #   -> use a packaged module (csv, text etc.) based on content of the directory
+    # - if path has no "/" and is a module on github (in /datasets or in /metrics)
+    #   -> use the module from the python file on github
+    # - if path has one "/" and is dataset repository on the HF hub with a python file
+    #   -> the module from the python file in the dataset repository
+    # - if path has one "/" and is dataset repository on the HF hub without a python file
+    #   -> use a packaged module (csv, text etc.) based on content of the repository
```

Comment:
Maybe this should be in the docstring or doc?

Reply:
It is, but the formulation is a bit different to be clearer for users.